tests/test_interfaces.py

```diff
@@ -778,6 +723,103 @@ def test_get_interface_mac_address(self):
 
         self.assertEqual(self.network_manager.get_interface_mac(interface_name), self.mac_address)
 
+    def test_get_interface_1_ap_1_mon_internet_interfaces(self):
```

Comment:
These tests are all wrong. Their name is wrong and I'm not sure why you are creating multiple interfaces to test `is_interface_valid` method. Also you are asserting `network_manager._active` variable as you answer. Private variables shouldn't be asserted as with a refactor they will fail.